### ISSUE TITLE
[Blazor WASM] SyntaxError: Unexpected end of JSON input

### ISSUE THREAD

**Issue by ScottKane** [dotnet/aspnetcore#27826]
### Describe the bug
I recently started upgrading a Blazor/.NET Core 3.1 project to .NET 5.0. I have changed all projects from either `netcoreapp3.1` or `netstandard2.1` to `net5.0`. I had to then switch namespaces from `Microsoft.AspNetCore.Blazor.Server`  to `Microsoft.AspNetCore.Components.WebAssembly.Server` and change:

`app.UseBlazorDebugging();` to `app.UseWebAssemblyDebugging();`
`app.UseClientSideBlazorFiles<Client.Program>();` to `app.UseBlazorFrameworkFiles();`
`endpoints.MapFallbackToClientSideBlazor<Client.Program>("index.html");` to `endpoints.MapFallbackToFile("index.html");`

Following these changes I get the following in the browser console:

```
_framework/blazor.boot.json:1 Failed to load resource: the server responded with a status of 404 ()
blazor.webassembly.js:1 SyntaxError: Unexpected end of JSON input
(anonymous) @ blazor.webassembly.js:1
```

### Further technical details
- OS: Windows x64
- .NET SDK: 5.0.100
- IDE: Visual Studio 2019 16.8.1

**Comment by ScottKane** [comment]
Ok I changed the target sdk from `<Project Sdk="Microsoft.NET.Sdk.Web">` to `<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">` and now I'm getting build errors in the client:

```
Error	MSB3030	Could not copy the file "D:\Repos\App\src\App.Client\obj\Debug\net5.0\compressed\.gz" because it was not found.	App.Client	C:\Users\scott\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\5.0.0-preview.6.20312.15\targets\StaticWebAssets.targets	61	
```

Currently have 494 of these messages for every referenced package for `dll.gz`, `dll.gz.gz` and `pbd.gz`

**Comment by MoryakSPb** [comment]
> 
> 
> I have updated to `<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Build" Version="5.0.0-preview.6.20312.15" PrivateAssets="all" />`
> 
> I also changed the target sdk from `<Project Sdk="Microsoft.NET.Sdk.Web">` to `<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">` and now I'm getting build errors in the client:
> 
> ```
> Error	MSB3030	Could not copy the file "D:\Repos\App\src\App.Client\obj\Debug\net5.0\compressed\.gz" because it was not found.	App.Client	C:\Users\scott\.nuget\packages\microsoft.aspnetcore.components.webassembly.build\5.0.0-preview.6.20312.15\targets\StaticWebAssets.targets	61	
> ```
> 
> Currently have 494 of these messages coming from the framework and referenced packages for the following file types:
> 
> ```
> dll.gz
> dll.gz.gz
> pbd.gz
> js.gz
> js.gz.gz
> blat.gz
> blat.gz.gz
> dat.gz
> dat.gz.gz
> ```

Remove the package reference to "Microsoft.AspNetCore.Components.WebAssembly.Build". It is no longer needed for build.
https://docs.microsoft.com/en-us/aspnet/core/migration/31-to-50

**Comment by ScottKane** [comment]
Thanks, I hadn't seen that article. I'm now getting the following in the browser console:

```
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.DataProtection.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Blazorise.Components.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Authentication.Core.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Authentication.Abstractions.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Authentication.Cookies.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Authentication.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Hosting.Server.Abstractions.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Hosting.Abstractions.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Identity.EntityFrameworkCore.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.WebUtilities.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Identity.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.AspNetCore.Http.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.EntityFrameworkCore.Abstractions.dll:1
Failed to load resource: the server responded with a status of 404 () :44306/_framework/_bin/Microsoft.EntityFrameworkCore.dll:1 
```

There's a lot more than this but from what I can tell, these files exist on disk "Client\bin\Debug\net5.0\wwwroot\_framework\_bin" but don't appear in the filesystem through the browser.

**Comment by ScottKane** [comment]
There is 715 errors in the console. If I add a package reference to my server project for DevServer the errors in the console is reduced to 496.

Server setup:
```
<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Server" Version="5.0.0" />
<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.0" PrivateAssets="all" />
```

Client setup:
```
<PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
```

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @ScottKane.
Consider creating a new app and using the references a new project template configures. 
That in addition to the migration docs should help your references sorted properly.
If there are still question we recommend using StackOverflow as this seems to be solely related to your current project.

**Comment by ScottKane** [comment]
@mkArtakMSFT why are you closing this issue? There is clearly an issue with the upgrade process from  a 3.1 Blazor WASM app to 5.0 when it is split into multiple projects. Please reopen the issue...

**Comment by rblanca** [comment]
Confirmed....Same here upgrading a 3.1 project to 5.0

**Comment by ScottKane** [comment]
@rblanca I raised a new issue to track this as this was closed #27910. Check to see if you have a publish profile in your server under the `Properties` folder. If you do, either remove it or check that it is targeting net5.0 instead of netcoreapp3.1. This was the issue for me.
